@fsharpasharp here I used file_groups it did avoid the package error for manifest and resources but I get the import error for `R`
https://github.com/oppia/oppia-android/blob/41ebb8693cf6d882d617081a1f99e843e916e591/utility/src/main/java/org/oppia/android/util/datetime/DateTimeUtil.kt#L4

```
ERROR: /home/farees/opensource/oppia-android/utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel:7:19: KotlinCompile //utility/src/main/java/org/oppia/android/util/datetime:date_time_util_kt { kt: 1, java: 0, srcjars: 0 } for k8 failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/io_bazel_rules_kotlin/src/main/kotlin/builder ... (remaining 1 argument(s) skipped)
utility/src/main/java/org/oppia/android/util/datetime/DateTimeUtil.kt:4:31: error: unresolved reference: R
import org.oppia.android.util.R
                              ^
Compilation failure: compile phase failed:
/home/farees/.cache/bazel/_bazel_farees/be6e0c0a55083b3b0484cd1169352d74/execroot/__main__/utility/src/main/java/org/oppia/android/util/datetime/DateTimeUtil.kt:4:31: error: unresolved reference: R
import org.oppia.android.util.R
                              ^
INFO: Elapsed time: 23.286s, Critical Path: 11.67s
INFO: 6 processes: 2 internal, 4 linux-sandbox.
FAILED: Build did NOT complete successfully

```

I guess this is similar to https://github.com/oppia/oppia-android/pull/2868#discussion_r591918000

--- a/utility/BUILD.bazel
+++ b/utility/BUILD.bazel
@@ -29,7 +29,7 @@ android_library(
     name = "strings",
     custom_package = "org.oppia.android.util",
     manifest = "src/main/AndroidManifest.xml",
-    resource_files = glob(["src/main/res/values/strings.xml"]),
+    resource_files = ["src/main/res/values/strings.xml"],
     visibility = ["//visibility:public"],
 )
 

--- a/utility/BUILD.bazel
+++ b/utility/BUILD.bazel
@@ -17,16 +17,8 @@ MIGRATED_PROD_FILES = glob([
     "src/main/java/org/oppia/android/util/system/*.kt",
 ])
 
-filegroup(
-    name = "manifest",
-    srcs = [
-        "src/main/AndroidManifest.xml",
-    ],
-    visibility = ["//visibility:public"],
-)
-
 android_library(
-    name = "strings",
+    name = "resources",
     custom_package = "org.oppia.android.util",
     manifest = "src/main/AndroidManifest.xml",
     resource_files = ["src/main/res/values/strings.xml"],
@@ -40,12 +32,10 @@ kt_android_library(
         ["src/main/java/org/oppia/android/util/**/*.kt"],
         exclude = MIGRATED_PROD_FILES,
     ),
-    custom_package = "org.oppia.android.util",
-    manifest = "src/main/AndroidManifest.xml",
-    resource_files = glob(["src/main/res/**/*.xml"]),
     visibility = ["//visibility:public"],
     deps = [
         ":dagger",
+        ":resources",
         "//app:crashlytics",
         "//app:crashlytics_deps",
         "//model",

--- a/utility/BUILD.bazel
+++ b/utility/BUILD.bazel
@@ -25,11 +25,11 @@ filegroup(
     visibility = ["//visibility:public"],
 )
 
-filegroup(
+android_library(
     name = "strings",
-    srcs = [
-        "src/main/res/values/strings.xml",
-    ],
+    custom_package = "org.oppia.android.util",
+    manifest = "src/main/AndroidManifest.xml",
+    resource_files = glob(["src/main/res/values/strings.xml"]),
     visibility = ["//visibility:public"],
 )
 

--- a/utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel
+++ b/utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel
@@ -12,7 +12,7 @@ kt_android_library(
     visibility = ["//:oppia_api_visibility"],
     deps = [
         "//third_party:javax_inject_javax_inject",
-        "//utility:strings",
+        "//utility:resources",
         "//utility/src/main/java/org/oppia/android/util/system:oppia_clock",
     ],
 )

--- a/utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel
+++ b/utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel
@@ -9,11 +9,10 @@ kt_android_library(
     srcs = [
         "DateTimeUtil.kt",
     ],
-    manifest = "//utility:manifest",
-    resource_files = ["//utility:strings"],
     visibility = ["//:oppia_api_visibility"],
     deps = [
         "//third_party:javax_inject_javax_inject",
+        "//utility:strings",
         "//utility/src/main/java/org/oppia/android/util/system:oppia_clock",
     ],
 )

